vendors/nordic/boards/nrf52840-dk/ports/common_io/iot_spi.c

```diff
@@ -0,0 +1,598 @@
+{
+    SpiContext_t const * pxSpiContext; /* Nordic Handle */
+    IotSPIMasterConfig_t xConfig;      /* Master Configuration */
+    IotSPICallback_t xSpiCallback;     /* Callback function */
+    void * pvUserContext;              /* User context passed in callback */
+    uint8_t sOpened;                   /* Bit flags to track different states. */
+} IotSPIDescriptor_t;
+/*-----------------------------------------------------------*/
+
+static const IotSPIMasterConfig_t xDefaultConfig =
+{
+    .ulFreq       = NRF_DRV_SPI_FREQ_4M,
+    .eMode        = eSPIMode0,
+    .eSetBitOrder = eSPIMSBFirst,
+    .ucDummyValue = 0xFF
+};
+
+static SpiContext_t xSpiContexts[] =
+{
+    {
+        .instance = NRF_DRV_SPI_INSTANCE( 0 ),
+        .config =
+        {
+            .sck_pin      = IOT_COMMON_IO_SPI_1_SCLK_PIN,
+            .mosi_pin     = IOT_COMMON_IO_SPI_1_MOSI_PIN,
+            .miso_pin     = IOT_COMMON_IO_SPI_1_MISO_PIN,
+            .ss_pin       = NRF_DRV_SPI_PIN_NOT_USED,
+            .irq_priority = SPI_DEFAULT_CONFIG_IRQ_PRIORITY,
+            .orc          = xDefaultConfig.ucDummyValue,
+            .frequency    = xDefaultConfig.ulFreq,
+            .mode         = xDefaultConfig.eMode,
+            .bit_order    = xDefaultConfig.eSetBitOrder == eSPIMSBFirst ? NRF_DRV_SPI_BIT_ORDER_MSB_FIRST : NRF_DRV_SPI_BIT_ORDER_LSB_FIRST,
+        }
+    },
+    {
+        .instance = NRF_DRV_SPI_INSTANCE( 1 ),
+        .config =
+        {
+            .sck_pin      = IOT_COMMON_IO_SPI_2_SCLK_PIN,
+            .mosi_pin     = IOT_COMMON_IO_SPI_2_MOSI_PIN,
+            .miso_pin     = IOT_COMMON_IO_SPI_2_MISO_PIN,
+            .ss_pin       = NRF_DRV_SPI_PIN_NOT_USED,
+            .irq_priority = SPI_DEFAULT_CONFIG_IRQ_PRIORITY,
+            .orc          = xDefaultConfig.ucDummyValue,
+            .frequency    = xDefaultConfig.ulFreq,
+            .mode         = xDefaultConfig.eMode,
+            .bit_order    = xDefaultConfig.eSetBitOrder == eSPIMSBFirst ? NRF_DRV_SPI_BIT_ORDER_MSB_FIRST : NRF_DRV_SPI_BIT_ORDER_LSB_FIRST,
+        }
+    },
+    {
+        .instance = NRF_DRV_SPI_INSTANCE( 2 ),
+        .config =
+        {
+            .sck_pin      = IOT_COMMON_IO_SPI_3_SCLK_PIN,
+            .mosi_pin     = IOT_COMMON_IO_SPI_3_MOSI_PIN,
+            .miso_pin     = IOT_COMMON_IO_SPI_3_MISO_PIN,
+            .ss_pin       = NRF_DRV_SPI_PIN_NOT_USED,
+            .irq_priority = SPI_DEFAULT_CONFIG_IRQ_PRIORITY,
+            .orc          = xDefaultConfig.ucDummyValue,
+            .frequency    = xDefaultConfig.ulFreq,
+            .mode         = xDefaultConfig.eMode,
+            .bit_order    = xDefaultConfig.eSetBitOrder == eSPIMSBFirst ? NRF_DRV_SPI_BIT_ORDER_MSB_FIRST : NRF_DRV_SPI_BIT_ORDER_LSB_FIRST,
+        }
+    }
+};
+
+/*
+ * Forward declare event handler passed to NRF SPI controller.
+ * Invokes user callbacks for async modes.
+ * Sync modes poll atomic flags to track state
+ */
+static volatile bool bTransferDone = true;
```

Comment:
Instead of global, this state should be associated with each SPI instance.

Reply:
Very true! Thank you for highlighting this.